widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx

```diff
@@ -156,62 +131,52 @@ export function ConfirmWalletsModal(props: PropTypes) {
-    !!customDestination &&
-    showCustomDestination &&
-    lastStepToBlockchain &&
-    !isValidAddress(lastStepToBlockchain, customDestination);
-
-  const resetCustomDestination = () => {
-    setShowCustomDestination(false);
-    setCustomDestination(null);
-    setSelectableWallets((selectableWallets) => {
-      let anyWalletSelected = false;
-      return selectableWallets.map((selectableWallet) => {
-        if (
-          !anyWalletSelected &&
-          selectableWallet.chain === lastStepToBlockchain?.name
-        ) {
-          anyWalletSelected = true;
-          return {
-            ...selectableWallet,
-            selected: true,
-          };
-        }
-        return selectableWallet;
-      });
-    });
-  };
-
-  const handleClearCustomDestination = () => setCustomDestination('');
-
-  const handlePasteCustomDestination = async (
-    event: MouseEvent<HTMLButtonElement>
+  const updateSelectableWallets = (
+    wallets: ConnectedWallet[],
+    chainName: string,
+    updateWallet: (wallet: ConnectedWallet) => ConnectedWallet
   ) => {
-    event.preventDefault();
-    if (navigator.clipboard !== undefined) {
-      const pastedText = await navigator.clipboard.readText();
-      setCustomDestination(pastedText);
-      customDestinationInputRef?.current?.focus();
-    }
+    return wallets.map((wallet) => {
+      if (wallet.chain === chainName) {
+        return updateWallet(wallet);
+      }
+      return wallet;
+    });
   };
-
   const handleCustomDestinationCollapsibleOpenChange = (checked: boolean) => {
+    setCustomDestinationOpen(checked);
     if (!checked) {
-      resetCustomDestination();
+      setSelectableWallets((selectableWallets) => {
+        let anyWalletSelected = false;
+        const updateWallet = (wallet: ConnectedWallet) => {
+          if (!anyWalletSelected) {
+            anyWalletSelected = true;
+            return {
+              ...wallet,
+              selected: true,
+            };
+          }
+          return wallet;
+        };
+        return updateSelectableWallets(
+          selectableWallets,
+          lastStepToBlockchain?.name || '',
+          updateWallet
+        );
+      });
     } else {
       if (!isWalletRequiredFor(lastStepToBlockchain?.name ?? '')) {
-        setSelectableWallets((selectableWallets) =>
-          selectableWallets.map((selectableWallet) => {
-            if (selectableWallet.chain === lastStepToBlockchain?.name) {
-              return {
-                ...selectableWallet,
-                selected: false,
-              };
-            }
-            return selectableWallet;
-          })
-        );
+        setSelectableWallets((selectableWallets) => {
+          const updateWallet = (wallet: ConnectedWallet) => ({
+            ...wallet,
+            selected: false,
+          });
+          return updateSelectableWallets(
+            selectableWallets,
+            lastStepToBlockchain?.name || '',
+            updateWallet
```

Comment:
`updateWallet` callbaack just sets `selected` property to `true` or `false`. Can't use just pass the `selected` value?

Reply:
fixed